### PR TITLE
Delete posts

### PR DESCRIPTION
**Pull request by Tirokk**
> [danielgrijalva](https://github.com/danielgrijalva) **Authored by [danielgrijalva](https://github.com/danielgrijalva)**
_<time datetime="2019-04-26T23:03:47Z" title="Saturday, April 27th 2019, 1:03:47 am +02:00">Apr 27, 2019</time>_
_Merged <time datetime="2019-05-10T14:56:01Z" title="Friday, May 10th 2019, 4:56:01 pm +02:00">May 10, 2019</time>_
---

## Pull Request
This lets users delete their own posts/contributions. 

- Created post deletion modal
- Added post deletion locales (EN and ES)
- Added "Delete post" option in contribution menu
- Redirects to `/` _or_ refreshes page after deleting a post

human connection delete posts demo

### Issues

- [X] Resolves issue #330 

### Todo

- [ ] Tests pending. They should be ready next week.
- [ ] Translations. Only english and spanish were provided.

### CODE REVIEW

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-02T17:33:52Z" title="Thursday, May 2nd 2019, 7:33:52 pm +02:00">May 2, 2019</time>_

---


@danielgrijalva @Tirokk I just resolved merge conflicts through Github.

We :heart: you @danielgrijalva

So much love

As you mentioned, tests are still missing. Translations are fine, we can add them through [lokalise](https://lokalise.co/project/556252725c18dd752dd546.13222042/).

@danielgrijalva @Tirokk would one of you be willing to contribute some frontend tests?

**Comment by Tirokk** [comment]
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-05-03T06:25:40Z" title="Friday, May 3rd 2019, 8:25:40 am +02:00">May 3, 2019</time>_

---


@roschaefer Best is @danielgrijalva write the Frontend tests himself. He seems to be smart to find out.

@danielgrijalva If you need help to write Frontend tests, please contact me ( @Tirokk ).
@mattwr18 Has a lot of experience in Frontend test writing too …

**Comment by Tirokk** [comment]
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-05-03T09:57:21Z" title="Friday, May 3rd 2019, 11:57:21 am +02:00">May 3, 2019</time>_

---


Are you on our (Human Connection Discord)[https://discord.gg/6ub73U3] chat @danielgrijalva ?

Would be nice to see you there! Please contact me (Wolle (Hamburg)) there, if you can.

**Comment by Tirokk** [comment]
> [danielgrijalva](https://github.com/danielgrijalva) **Authored by [danielgrijalva](https://github.com/danielgrijalva)**
_<time datetime="2019-05-03T17:45:58Z" title="Friday, May 3rd 2019, 7:45:58 pm +02:00">May 3, 2019</time>_

---


@Tirokk I'd be glad to provide the tests! 😄 but it'll take some more time... you see, I injured my right hand in an accident, so I won't be able to code for a week or so. I'll join the discord and contact you as soon as I can 🙆‍♂

**Comment by Tirokk** [comment]
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-05-04T07:42:12Z" title="Saturday, May 4th 2019, 9:42:12 am +02:00">May 4, 2019</time>_

---


Thanks for response @danielgrijalva !

I wish you a good recovery and hope that it hurts not to much and your hand gets quick well again.
Take the time you need and I am happy to hear you.
See you on Discord.

File: webapp/components/Modal/DeleteModal.spec.js
Comment (Tirokk) [inline webapp/components/Modal/DeleteModal.spec.js:14]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-08T21:17:06Z" title="Wednesday, May 8th 2019, 11:17:06 pm +02:00">May 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,155 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DeleteModal from './DeleteModal.vue'
+import Vue from 'vue'
+import Vuex from 'vuex'
+import Styleguide from '@human-connection/styleguide'
+import VueRouter from 'vue-router'
+
+const routes = [{ path: '/' }]
+const router = new VueRouter({routes})
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(Styleguide)
+localVue.use(VueRouter)
```


using this `VueRouter` in the tests is throwing these errors
Screenshot at 2019-05-08 18:13:21

Comment (Tirokk) [reply webapp/components/Modal/DeleteModal.spec.js:14]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-08T21:19:12Z" title="Wednesday, May 8th 2019, 11:19:12 pm +02:00">May 8, 2019</time>_

---


if you want to read more, I found this issue
https://github.com/jsdom/jsdom/issues/2112

File: webapp/components/Modal/DeleteModal.spec.js
Comment (Tirokk) [inline webapp/components/Modal/DeleteModal.spec.js:38]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-08T21:19:18Z" title="Wednesday, May 8th 2019, 11:19:18 pm +02:00">May 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,155 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DeleteModal from './DeleteModal.vue'
+import Vue from 'vue'
+import Vuex from 'vuex'
+import Styleguide from '@human-connection/styleguide'
+import VueRouter from 'vue-router'
+
+const routes = [{ path: '/' }]
+const router = new VueRouter({routes})
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(Styleguide)
+localVue.use(VueRouter)
+
+describe('DeleteModal.vue', () => {
+  let wrapper
+  let Wrapper
+  let propsData
+  let mocks
+
+  beforeEach(() => {
+    propsData = {
+      type: 'contribution',
+      id: 'c300'
+    }
+    mocks = {
+      $t: jest.fn(),
+      $filters: {
+        truncate: a => a
+      },
+      $toast: {
+        success: () => {},
+        error: () => {}
+      },
+      $apollo: {
+        mutate: jest.fn().mockResolvedValue()
+      }
```


if you add this to the `mocks`
```
$router: {
        history: {
          push: jest.fn(),
          current: {
            name: 'post-id-slug'
          }
        }
      }
```
your tests still pass, but there are no errors

File: webapp/components/Modal/DeleteModal.spec.js
Comment (Tirokk) [inline webapp/components/Modal/DeleteModal.spec.js:85]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-08T21:44:02Z" title="Wednesday, May 8th 2019, 11:44:02 pm +02:00">May 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,155 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DeleteModal from './DeleteModal.vue'
+import Vue from 'vue'
+import Vuex from 'vuex'
+import Styleguide from '@human-connection/styleguide'
+import VueRouter from 'vue-router'
+
+const routes = [{ path: '/' }]
+const router = new VueRouter({routes})
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(Styleguide)
+localVue.use(VueRouter)
+
+describe('DeleteModal.vue', () => {
+  let wrapper
+  let Wrapper
+  let propsData
+  let mocks
+
+  beforeEach(() => {
+    propsData = {
+      type: 'contribution',
+      id: 'c300'
+    }
+    mocks = {
+      $t: jest.fn(),
+      $filters: {
+        truncate: a => a
+      },
+      $toast: {
+        success: () => {},
+        error: () => {}
+      },
+      $apollo: {
+        mutate: jest.fn().mockResolvedValue()
+      }
+    }
+  })
+
+  describe('shallowMount', () => {
+    const Wrapper = () => {
+      return shallowMount(DeleteModal, { propsData, mocks, localVue, router })
+    }
+
+    describe('defaults', () => {
+      it('success false', () => {
+        expect(Wrapper().vm.success).toBe(false)
+      })
+
+      it('loading false', () => {
+        expect(Wrapper().vm.loading).toBe(false)
+      })
+    })
+    
+    describe('given a post', () => {
+      beforeEach(() => {
+        propsData = {
+          id: 'p23',
+          type: 'post',
+          name: 'It is a post'
+        }
+      })
+
+      it('mentions post title', () => {
+        Wrapper()
+        const calls = mocks.$t.mock.calls
+        const expected = [['post.delete.message', { name: 'It is a post' }]]
+        expect(calls).toEqual(expect.arrayContaining(expected))
+      })
+    })
+  })
+
+  describe('mount', () => {
+    const Wrapper = () => {
+      return mount(DeleteModal, { propsData, mocks, localVue, router})
+    }
+
+    beforeEach(jest.useFakeTimers)
+
+    it('renders', () => {
+      expect(Wrapper().is('div')).toBe(true)
+    })
+
```


I know there are many examples of this in the code base, but I am not a fan of these `renders` tests
I think we could open a ticket to remove them @roschaefer, to be honest. 
If the component does not render, then all the other tests will fail and therefore are testing that the component renders already.
It might not take a long time to run these tests every build in all, or most of the test files, but it definitely does take time and I don't see what benefit they provide.
Nothing that I think should hold up your `PR` getting merged though @danielgrijalva, just something for us to discuss as a team.

File: webapp/components/Modal/DeleteModal.spec.js
Comment (Tirokk) [inline webapp/components/Modal/DeleteModal.spec.js:109]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-08T21:52:40Z" title="Wednesday, May 8th 2019, 11:52:40 pm +02:00">May 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,155 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DeleteModal from './DeleteModal.vue'
+import Vue from 'vue'
+import Vuex from 'vuex'
+import Styleguide from '@human-connection/styleguide'
+import VueRouter from 'vue-router'
+
+const routes = [{ path: '/' }]
+const router = new VueRouter({routes})
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(Styleguide)
+localVue.use(VueRouter)
+
+describe('DeleteModal.vue', () => {
+  let wrapper
+  let Wrapper
+  let propsData
+  let mocks
+
+  beforeEach(() => {
+    propsData = {
+      type: 'contribution',
+      id: 'c300'
+    }
+    mocks = {
+      $t: jest.fn(),
+      $filters: {
+        truncate: a => a
+      },
+      $toast: {
+        success: () => {},
+        error: () => {}
+      },
+      $apollo: {
+        mutate: jest.fn().mockResolvedValue()
+      }
+    }
+  })
+
+  describe('shallowMount', () => {
+    const Wrapper = () => {
+      return shallowMount(DeleteModal, { propsData, mocks, localVue, router })
+    }
+
+    describe('defaults', () => {
+      it('success false', () => {
+        expect(Wrapper().vm.success).toBe(false)
+      })
+
+      it('loading false', () => {
+        expect(Wrapper().vm.loading).toBe(false)
+      })
+    })
+    
+    describe('given a post', () => {
+      beforeEach(() => {
+        propsData = {
+          id: 'p23',
+          type: 'post',
+          name: 'It is a post'
+        }
+      })
+
+      it('mentions post title', () => {
+        Wrapper()
+        const calls = mocks.$t.mock.calls
+        const expected = [['post.delete.message', { name: 'It is a post' }]]
+        expect(calls).toEqual(expect.arrayContaining(expected))
+      })
+    })
+  })
+
+  describe('mount', () => {
+    const Wrapper = () => {
+      return mount(DeleteModal, { propsData, mocks, localVue, router})
+    }
+
+    beforeEach(jest.useFakeTimers)
+
+    it('renders', () => {
+      expect(Wrapper().is('div')).toBe(true)
+    })
+
+    describe('given id', () => {
+      beforeEach(() => {
+        propsData = {
+          type: 'user',
+          id: 'u3'
+        }
+        wrapper = Wrapper()
+      })
+
+      describe('click cancel button', () => {
+        beforeEach(() => {
+          wrapper = Wrapper()
+          wrapper.find('button.cancel').trigger('click')
+        })
+
+        describe('after timeout', () => {
+          beforeEach(jest.runAllTimers)
+
+          it('fades away', () => {
+            expect(wrapper.vm.isOpen).toBe(false)
+          })
+
+          it('emits "close"', () => {
+            expect(wrapper.emitted().close).toBeTruthy()
```


I am not a fan of these tests, which test `toBeTruthy()` because they can be flaky...
could we test `expect(wrapper.emitted().close.length).toBe(1)`?
here is something on the subject, if you want to have a look https://stackoverflow.com/questions/32615713/tobetrue-vs-tobetruthy-vs-tobetrue

Again, I know there are many places in the code base that use this, and it is even given as an example in the vue-test-utils docs, but to me this would improve this test case
https://vue-test-utils.vuejs.org/api/wrapper/#emitted

File: webapp/components/Modal/DeleteModal.spec.js
Comment (Tirokk) [inline webapp/components/Modal/DeleteModal.spec.js:124]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-08T21:54:05Z" title="Wednesday, May 8th 2019, 11:54:05 pm +02:00">May 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,155 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DeleteModal from './DeleteModal.vue'
+import Vue from 'vue'
+import Vuex from 'vuex'
+import Styleguide from '@human-connection/styleguide'
+import VueRouter from 'vue-router'
+
+const routes = [{ path: '/' }]
+const router = new VueRouter({routes})
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(Styleguide)
+localVue.use(VueRouter)
+
+describe('DeleteModal.vue', () => {
+  let wrapper
+  let Wrapper
+  let propsData
+  let mocks
+
+  beforeEach(() => {
+    propsData = {
+      type: 'contribution',
+      id: 'c300'
+    }
+    mocks = {
+      $t: jest.fn(),
+      $filters: {
+        truncate: a => a
+      },
+      $toast: {
+        success: () => {},
+        error: () => {}
+      },
+      $apollo: {
+        mutate: jest.fn().mockResolvedValue()
+      }
+    }
+  })
+
+  describe('shallowMount', () => {
+    const Wrapper = () => {
+      return shallowMount(DeleteModal, { propsData, mocks, localVue, router })
+    }
+
+    describe('defaults', () => {
+      it('success false', () => {
+        expect(Wrapper().vm.success).toBe(false)
+      })
+
+      it('loading false', () => {
+        expect(Wrapper().vm.loading).toBe(false)
+      })
+    })
+    
+    describe('given a post', () => {
+      beforeEach(() => {
+        propsData = {
+          id: 'p23',
+          type: 'post',
+          name: 'It is a post'
+        }
+      })
+
+      it('mentions post title', () => {
+        Wrapper()
+        const calls = mocks.$t.mock.calls
+        const expected = [['post.delete.message', { name: 'It is a post' }]]
+        expect(calls).toEqual(expect.arrayContaining(expected))
+      })
+    })
+  })
+
+  describe('mount', () => {
+    const Wrapper = () => {
+      return mount(DeleteModal, { propsData, mocks, localVue, router})
+    }
+
+    beforeEach(jest.useFakeTimers)
+
+    it('renders', () => {
+      expect(Wrapper().is('div')).toBe(true)
+    })
+
+    describe('given id', () => {
+      beforeEach(() => {
+        propsData = {
+          type: 'user',
+          id: 'u3'
+        }
+        wrapper = Wrapper()
+      })
+
+      describe('click cancel button', () => {
+        beforeEach(() => {
+          wrapper = Wrapper()
+          wrapper.find('button.cancel').trigger('click')
+        })
+
+        describe('after timeout', () => {
+          beforeEach(jest.runAllTimers)
+
+          it('fades away', () => {
+            expect(wrapper.vm.isOpen).toBe(false)
+          })
+
+          it('emits "close"', () => {
+            expect(wrapper.emitted().close).toBeTruthy()
+          })
+
+          it('does not call mutation', () => {
+            expect(mocks.$apollo.mutate).not.toHaveBeenCalled()
+          })
+        })
+      })
+
+      describe('click confirm button', () => {
+        beforeEach(() => {
+          wrapper.find('button.confirm').trigger('click')
+        })
+
+        it('calls delete mutation', () => {
+          expect(mocks.$apollo.mutate).toHaveBeenCalled()
```


maybe we could use `toHaveBeenCalledTimes(1)`?

File: webapp/components/Modal/DeleteModal.spec.js
Comment (Tirokk) [reply webapp/components/Modal/DeleteModal.spec.js:85]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-08T23:35:59Z" title="Thursday, May 9th 2019, 1:35:59 am +02:00">May 9, 2019</time>_

---


@mattwr18 sure, usually I start off by writing this kind of test case only to make sure I have everything set in place to start the component test. Later, I just don't remove it. I won't miss it so badly, so go ahead removing it.

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-09T10:34:31Z" title="Thursday, May 9th 2019, 12:34:31 pm +02:00">May 9, 2019</time>_

---


@roschaefer @danielgrijalva is in UTC -7, so I think the meetings would be in the middle of the night for him.

We have talked about having some pairing sessions around `9pm UTC`, but so far we haven't hooked up.

As I stated :point_up:, I also think it's a great job, and was giving suggestions, not blocking it being merged with requested changes. :smile:

**Comment by Tirokk** [comment]
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-09T23:36:42Z" title="Friday, May 10th 2019, 1:36:42 am +02:00">May 10, 2019</time>_

---


@roschaefer is here a problem regarding travis since the branch is on a foreign repo? If yes i would be very interested in the solution

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-10T11:19:40Z" title="Friday, May 10th 2019, 1:19:40 pm +02:00">May 10, 2019</time>_

---


@ulfgebhardt I observed the same: If we receive PR from `origin`<-`fork` then the `Travis CI Branch` check is pending indefinitely. I have 0 idea why. Could you do some research?

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-10T11:52:17Z" title="Friday, May 10th 2019, 1:52:17 pm +02:00">May 10, 2019</time>_

---


Maybe, irrelevant, but just as a reminder, and also for @ulfgebhardt who wasn't around on a daily basis when we had the discussion, encrypted variables in Travis are not accessible by `forks`.

Again, maybe not relevant as I think @ulfgebhardt you said that we are not using `$CODE_COV`, but at the time we had the discussion about encouraging people not to fork and put in `PR`s, but to ask, and we would happily give them write access.

**Comment by Tirokk** [comment]
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-10T14:34:41Z" title="Friday, May 10th 2019, 4:34:41 pm +02:00">May 10, 2019</time>_

---


i will just push this on our repo and merge - noting the research task in order to create a ticket

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-05-10T14:52:53Z" title="Friday, May 10th 2019, 4:52:53 pm +02:00">May 10, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511?src=pr&el=h1) Report
> Merging [#511](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511?src=pr&el=desc) into [master](https://codecov.io/gh/Human-Connection/Human-Connection/commit/a9860c49fc3f717b8694bc12f73258f253c045f4?src=pr&el=desc) will **increase** coverage by `1.29%`.
> The diff coverage is `78.57%`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #511      +/-   ##
==========================================
+ Coverage   16.86%   18.15%   +1.29%     
==========================================
  Files         154      155       +1     
  Lines        2147     2231      +84     
  Branches       92      112      +20     
==========================================
+ Hits          362      405      +43     
- Misses       1765     1803      +38     
- Partials       20       23       +3
```


| [Impacted Files](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [webapp/components/ContentMenu.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvQ29udGVudE1lbnUudnVl) | `36% <0%> (-3.13%)` | :arrow_down: |
| [webapp/components/Modal.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvTW9kYWwudnVl) | `76.92% <100%> (+1.92%)` | :arrow_up: |
| [webapp/components/Modal/DeleteModal.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvTW9kYWwvRGVsZXRlTW9kYWwudnVl) | `84% <84%> (ø)` | |
| [webapp/components/User/index.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvVXNlci9pbmRleC52dWU=) | `92.31% <0%> (-7.69%)` | :arrow_down: |
| [backend/src/middleware/userMiddleware.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvbWlkZGxld2FyZS91c2VyTWlkZGxld2FyZS5qcw==) | `0% <0%> (ø)` | :arrow_up: |
| [webapp/layouts/default.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511/diff?src=pr&el=tree#diff-d2ViYXBwL2xheW91dHMvZGVmYXVsdC52dWU=) | `2.86% <0%> (+2.86%)` | :arrow_up: |
| [webapp/pages/profile/\_id/\_slug.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511/diff?src=pr&el=tree#diff-d2ViYXBwL3BhZ2VzL3Byb2ZpbGUvX2lkL19zbHVnLnZ1ZQ==) | `5.26% <0%> (+5.26%)` | :arrow_up: |
| [webapp/pages/settings/index.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/511/diff?src=pr&el=tree#diff-d2ViYXBwL3BhZ2VzL3NldHRpbmdzL2luZGV4LnZ1ZQ==) | `29.63% <0%> (+29.63%)` | :arrow_up: |

**Comment by Tirokk** [comment]
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-10T14:55:27Z" title="Friday, May 10th 2019, 4:55:27 pm +02:00">May 10, 2019</time>_

---


@roschaefer the only solution to this travis bug is to disable branch build (we should discuss this) - a workaround is to finish everything in the forks pr and then push the branch to the HC Repo. The approves will be kept. Can be seen here: https://github.com/Human-Connection/Human-Connection/pull/595

Ressource: https://travis-ci.community/t/travis-ci-branch-expected-waiting-for-status-to-be-reported/829

It also seems like the connected issue on the Forks-Branch-PR is not closed when merging the HC-Branch-PR. Even tho both PRs get `merged` status

**Review by Mogge — APPROVED** [review]
Hey @danielgrijalva ,

this is wonderful work, as far I can see. Well done !!! 🚀💫

There is one little merge conflict with the master left, but nothing serious. 🤗
And the Frontend tests have to be implemented for sure, like you already said.

Thanks you!

**Review by Mogge — APPROVED** [review]
@danielgrijalva as @mattwr18 said, a fullstack test would be nice. We use fullstack and :cucumber: tests to document behaviour that reaches the end-user.

However I'll **approve** this pull request. I find it already quite awesome and for me, it's OK to if sb. of the core contributors take over the fullstack test duties. Thank you so much @danielgrijalva. Maybe you want to join one of our pair-programmings or maybe tomorrow in our weekly Open-Source community meetings? (UTC 11am, link will be sent on Discord)

**Pull request closed without merging**
